src/common/event-emitter.ts

```diff
@@ -52,12 +48,14 @@ export class EventEmitter<D extends [...any[]]> {
   }
 
   emit(...data: D) {
-    [...this.listeners].every(([callback, options]) => {
```

Comment:
I agree that .every is a misuse but will be great if adding tests to ensure .emit do the same before and after refactoring.

Reply:
I will add tests